lib/network/modules/components/physics/CentralGravitySolver.js

```diff
@@ -1,12 +1,30 @@
+'use strict'
+
+var util = require('../../../../util');
+
 class CentralGravitySolver {
-  constructor(body, physicsBody, options) {
+  constructor(body, physicsBody, options, groups) {
     this.body = body;
     this.physicsBody = physicsBody;
+    this.groups = groups;
+    this.gravityGroups = {};
     this.setOptions(options);
   }
 
   setOptions(options) {
     this.options = options;
+
+    // Iterates through the groups and looks for the ones defining 'groupCentralGravity'
+    if (this.groups != null) {
+      self = this;
+      util.forEach(this.groups.groups, function(groupDefinition, groupName) {
```

Comment:
this way of iteration is much slower than the for in way and you won't need self = this